core/coreapi/unixfs.go

```diff
@@ -196,12 +213,6 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.Node, opts ...options
 		return path.ImmutablePath{}, err
 	}
 
-	if !settings.OnlyHash {
-		if err := api.provider.Provide(ctx, nd.Cid(), true); err != nil {
-			return path.ImmutablePath{}, err
-		}
-	}
-
```

Comment:
Did this not happen from the providing.Exchange already? This might have been a leftover from long ago. Traceable to https://github.com/ipfs/kubo/commit/a9b65346871e845a67bbd19c668b699c55095e41.

Anyways, no need to do this here as we will respect the strategy.